### ISSUE TITLE
Incorrect indices of matched characters

### ISSUE THREAD

**Issue by cslee** [krisk/Fuse#735]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the bug

`indices` returned in below example contains overlapped positions

```
const Fuse = require('fuse.js')

const list = [
  {
    title: '/images/payment/select-payment-icon.png',
  },
]

const fuse = new Fuse(list, {
  includeMatches: true,
  keys: ['title'],
})

const result = fuse.search('images-payment-select-payment-icon')

console.log(result[0].matches[0].indices)
// Output:
// [
//   [1, 6],
//   [8, 14],
//   [16, 32],
//   [34, 34],
//   [36, 38],
//   [1, 6],
//   [8, 14],
//   [16, 34],
//   [36, 38],
// ]
```

### The Fuse.js version where this bug is happening.

6.6.2

### Is this a regression?

- [ ] This is a regression bug

### Which version did this behavior use to work in?

None

### Steps To Reproduce

Run the script in the description

### Expected behavior

`indices` should be matched characters starting from the first one like this:

```
console.log(result[0].matches[0].indices)
// Output:
// [
//   [1, 6],
//   [8, 14],
//   [16, 34],
//   [36, 38],
// ]
```

**Comment by 9mm** [comment]
I also have fuse highlighting like tons of letters that I havent even typed...

**Comment by hofmannjan1** [comment]
+1

**Comment by krisk** [comment]
Fixed in 60c393a4. Match indices from chunked patterns (>32 chars) are now sorted and merged into non-overlapping ranges, eliminating duplicates.
